pymc/aesaraf.py

```diff
@@ -537,23 +537,64 @@ def join_nonshared_inputs(
     point: Dict[str, np.ndarray],
     xs: List[TensorVariable],
     vars: List[TensorVariable],
-    shared,
+    shared: Dict[TensorVariable, TensorSharedVariable],
     make_shared: bool = False,
-):
+) -> Tuple[List[TensorVariable], TensorVariable]:
     """
-    Takes a list of Aesara Variables and joins their non shared inputs into a single input.
+    Takes a list of TensorVariables and joins their non shared inputs into a single input.
```

Comment:
```suggestion
    Takes a list of TensorVariables and joins their non shared inputs into a single raveled vector input.
```

Reply:
I incorporated this but included bit about the returned outputs as well. Let me know how it sounds